metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py

```diff
@@ -482,6 +496,10 @@ def on_task_instance_running(
     def on_task_instance_finish(
         self, task_instance: "TaskInstance", status: InstanceRunResult
     ) -> None:
+        if self.check_kill_switch():
```

Comment:
this method is called by the success and failure hooks - so it doesn't need this